aspnetcore/test/integration-tests/samples/2.x/IntegrationTestsSample/tests/RazorPagesProject.Tests/IntegrationTests/AuthTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using AngleSharp.Dom.Html;
+using Xunit;
+using RazorPagesProject.Services;
+using RazorPagesProject.Tests.Helpers;
+
+namespace RazorPagesProject.Tests
+{
+    public class AuthTests : 
+        IClassFixture<CustomWebApplicationFactory<RazorPagesProject.Startup>>
+    {
+        private readonly CustomWebApplicationFactory<RazorPagesProject.Startup> 
+            _factory;
+
+        public AuthTests(
+            CustomWebApplicationFactory<RazorPagesProject.Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        #region snippet1
+        [Fact]
+        public async Task Get_GithubProfilePageCanGetAGithubUser()
+        {
+            // Arrange
+            void ConfigureTestServices(IServiceCollection services) =>
+                services.AddSingleton<IGithubClient>(new TestGithubClient());
+            var client = _factory
+                .WithWebHostBuilder(builder => 
+                    builder.ConfigureTestServices(ConfigureTestServices))
+                .CreateClient();
+
+            // Act
+            var profile = await client.GetAsync("/GithubProfile");
+            Assert.Equal(HttpStatusCode.OK, profile.StatusCode);
+            var profileHtml = await HtmlHelpers.GetDocumentAsync(profile);
+
+            var profileWithUserName = await client.SendAsync(
+                (IHtmlFormElement)profileHtml.QuerySelector("#user-profile"),
+                new Dictionary<string, string> { ["Input_UserName"] = "user" });
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, profileWithUserName.StatusCode);
+            var profileWithUserHtml = 
+                await HtmlHelpers.GetDocumentAsync(profileWithUserName);
+            var userLogin = profileWithUserHtml.QuerySelector("#user-login");
+            Assert.Equal("user", userLogin.TextContent);
+        }
+
+        public class TestGithubClient : IGithubClient
+        {
+            public Task<GithubUser> GetUserAsync(string userName)
+            {
+                if (userName == "user")
+                {
+                    return Task.FromResult(
+                        new GithubUser
```

Comment:
Consider putting these on one line.

Reply:
Can't because we don't want horizontal scrollbars in the docs.